x/concentrated-liquidity/lp.go

```diff
@@ -250,6 +250,57 @@ func (k Keeper) withdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 	return actualAmount0.Neg(), actualAmount1.Neg(), nil
 }
 
+// nolint: unused
+// addToPosition attempts to add liquidityAmount to a position with the given pool id in the given tick range.
+// Returns error if
+// - Withdrawing full position fails
+// - Creating new position with added liquidity fails
+// - Position is superfluid staked
+// TODO: handle adding to SFS positions
+func (k Keeper) addToPosition(ctx sdk.Context, owner sdk.AccAddress, positionId uint64, amount0Added, amount1Added sdk.Int) (uint64, sdk.Int, sdk.Int, error) {
+	position, err := k.GetPosition(ctx, positionId)
+	if err != nil {
+		return 0, sdk.Int{}, sdk.Int{}, err
+	}
+
+	if amount0Added.IsNegative() || amount1Added.IsNegative() {
+		return 0, sdk.Int{}, sdk.Int{}, types.NegativeAmountAddedError{PositionId: position.PositionId, Asset0Amount: amount0Added, Asset1Amount: amount1Added}
+	}
+
+	// If the position is superfluid staked, return error.
+	// TODO: handle this case to allow LPs to add to SFS positions
+	positionHasUnderlyingLock, err := k.positionHasUnderlyingLockInState(ctx, positionId)
```

Comment:
Note on this, I think I am going to change this to "positionHasActiveUnderlyingLockInState", so the method will not only check if there is a lock state entry, but also check that that lock still exists